### DISCUSSION TITLE
How to handle data transfer during node fail-over using BBoltDB

### DISCUSSION THREAD

**Discussion by aarthikrao** [aarthikrao/timeMachine#5] [Q&A]
**Problem**
When a node is temporarily unavailable due to a temporary network partition, it will not contain the latest commited data. It will have to catch up to the leader vnode. How do we achieve this without compromising on service availability.

**Quick solution**
Append-only log file. This is supported by the LSM tree. Need to check for B-tree based BBoltDB

**Comment by devansh42** [comment]
Can give you more context on this problem?

**Reply by aarthikrao** [reply]
Updated
